Sources/StreamVideo/WebRTC/WebRTCClient.swift

```diff
@@ -1163,10 +1201,93 @@ class WebRTCClient: NSObject, @unchecked Sendable {
         )
     }
     
+    private func subscribeToInternetConnectionUpdates() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleConnectionStateChange),
+            name: .internetConnectionStatusDidChange,
+            object: nil
+        )
+    }
+    
+    @objc private func handleConnectionStateChange(_ notification: NSNotification) {
+        guard let status = notification.userInfo?[Notification.internetConnectionStatusUserInfoKey] as? InternetConnection.Status else {
+            return
+        }
+        
+        if status == .unavailable {
+            if !isFastReconnecting {
+                isFastReconnecting = true
+            }
+            return
+        }
+        if status.isAvailable && isFastReconnecting, let url = signalChannel?.connectURL {
+            signalChannel = makeWebSocketClient(
+                url: url,
+                apiKey: .init(apiKey),
+                isFastReconnect: true
+            )
```

Comment:
Do we need to update the signalChannel instance? Isn't there a way to request it to reconnect?

Reply:
When you switch networks, your IP address changes. So all the TCP connections that the device has breaks. So when you do a fast reconnect, you have to re-establish a new WebSocket connection. 

This is because a TCP connection is identified by a 4-tuple which is (sourceIP, sourcePort, destinationIP, destinationPort). On network change, source IP and port change and the TCP connection is no longer valid.